tests/extmod/vfs_blockdev_invalid.py

```diff
@@ -17,9 +17,16 @@ def __init__(self, blocks):
         self.data = bytearray(blocks * self.ERASE_BLOCK_SIZE)
         self.read_res = 0
         self.write_res = 0
+        self.readblocks_history = []
 
     def readblocks(self, block, buf, off=0):
-        print("readblocks")
+        # FAT filesystem access may insert an additional read to blocks 0 and 2 in
+        # certain build configurations.  Don't print out those accesses to ensure that
+        # the test output is consistent across targets.
+        self.readblocks_history.append(block)
+        if self.readblocks_history not in ([0, 0, 2, 0], [0, 0, 2, 0, 2]):
+            print("readblocks", block, len(buf), off)
```

Comment:
I know I wrote this test to `print("readblocks")` here but I'm not sure if it's really necessary for anything this test verifies. Plus it couples the test very strongly to the implementation details of the filesystem, making it brittle (as per this issue).

I reckon it might be enough to replace all of this with a commented out print statement, that can be un-commented for debugging. What do you think?

Reply:
I added a commit to comment out the print (will squash before merge).

Retested on PYBD-SF2, Pico, ESP32_GENERIC.